src/os_posix/os_fs.c

```diff
@@ -35,9 +35,7 @@
  * the file system may not know that the data was written, so it may not flush the metadata
  * prior to data being written. Therefore, the block may be reallocated or lost upon crash.
  *
- * WiredTiger currently disallows using the mmap option with the direct I/O option. We are relying
- * on the user correctly specifying the direct I/O option if they mount a file system as DAX. If
- * we did not wish to rely on the user supplying the correct flags, we have two options:
+ * If we do not wish to rely on the user supplying the correct flags for mmap, we have two options:
```

Comment:
Hmm, this is interesting - it might need to go deeper than a comment change. Either way, the comment needs to be altered - there is no longer a flag that the user can provide to make sure this behaves correctly, so really we have two options for working around the problem (and the user doesn't enter into the equation).

From the comment it sounds like we have indeed solved the problem by implementing (2). If that's the case, we can probably lose this paragraph completely, and present (1) as an alternative to (2).

Reply:
Perhaps it can just be reworked slightly to say: ... `we do not rely on the user supply` ...
And at the bottom of this comment add a sentence that says: `This implements (2).`

Reply:
I have pinged @fedorova about this comment and will wait for a response before attempting to rewrite.

Reply:
It is safe to delete this comment altogether. WiredTiger allows using mmap in two ways: The first option (mmap) allows mapping read-only files: Use memory mapping when accessing files in a read-only mode. In that case, the issue described in the comment, where the client writes something to the file without the file system knowing it, does not apply.   The second way is via mmap_all option. The issue does would not occur in that situation  either. mmap_all maps the file, but does not write bytes to it directly. It only writes via __posix_file_write_mmap, which will grow default to making the write system call if the file needs to be grown and will then remap the file.

Reply:
Thanks Sasha, I have removed the comment here [05eeafa](https://github.com/wiredtiger/wiredtiger/pull/10809/commits/05eeafa8c523e7e60b35796ef9b0d870e8eb1624)